components/eamxx/src/physics/mam/mam_coupling.hpp

```diff
@@ -585,8 +590,30 @@ void compute_vertical_layer_heights(const Team& team,
     "Given column index does not correspond to given team!");
 
   const auto dz = ekat::subview(dry_atm.dz, column_index);
-  auto z_iface  = ekat::subview(dry_atm.z_iface, column_index);
-  auto z_mid    = ekat::subview(dry_atm.z_mid, column_index);
+  const auto z_iface  = ekat::subview(dry_atm.z_iface, column_index);
+  const auto z_mid    = ekat::subview(dry_atm.z_mid, column_index);
+  PF::calculate_z_int(team, mam4::nlev, dz, dry_atm.z_surf, z_iface);
+  team.team_barrier(); // likely necessary to have z_iface up to date
+  PF::calculate_z_mid(team, mam4::nlev, z_iface, z_mid);
+}
+
+KOKKOS_INLINE_FUNCTION
+void compute_vertical_layer_heights(const Team& team,
+                                    const DryAtmosphere& dry_atm,
+                                    const WetAtmosphere &wet_aero,
```

Comment:
There is a typo here, it should be `wet_atm` instead of `wet_aero`. Also, we may need to rearrange calls in the preprocess operator so that we can use `dry_atm` for all computations. That is, we need to compute `dry_atm` before we call `compute_vertical_layer_heights` in the preprocess operator.

Reply:
I removed `const WetAtmosphere &wet_aero` and used `dry_atm` to compute dz`.